sdk/identity/azure-identity/azure/identity/_internal/shared_token_cache.py

```diff
@@ -88,13 +89,19 @@ class SharedTokenCacheBase(ABC):
     def __init__(self, username=None, **kwargs):  # pylint:disable=unused-argument
         # type: (Optional[str], **Any) -> None
 
-        authority = kwargs.pop("authority", None)
-        self._authority = normalize_authority(authority) if authority else get_default_authority()
-
-        environment = urlparse(self._authority).netloc
-        self._environment_aliases = KNOWN_ALIASES.get(environment) or frozenset((environment,))
-        self._username = username
-        self._tenant_id = kwargs.pop("tenant_id", None)
+        self._auth_record = kwargs.pop("authentication_record", None)  # type: Optional[AuthenticationRecord]
+        if self._auth_record:
+            self._authority = self._auth_record.authority
+            self._username = self._auth_record.username
+            self._tenant_id = self._auth_record.tenant_id
+            self._environment_aliases = frozenset((self._authority,))
```

Comment:
I expect w/ or w/o auth_record, we should have same behavior here as

environment = urlparse(self._authority).netloc
self._environment_aliases = KNOWN_ALIASES.get(environment) or frozenset((environment,))
?

Reply:
Good eye. We don't need to parse the netloc out of the record's `authority` because the record's authority is already a netloc. As for why I'm ignoring potential aliases: the purpose of the record is to enable an application to use tokens it cached in prior authentications. An `AuthenticationRecord` and the account it represents will have identical values for `authority` because they're both results of the same authentication. An account having an alias of the record's `authority` was cached during a different authentication, and therefore doesn't match the record.